docs/poly-apis.md

```diff
@@ -0,0 +1,636 @@
+
+
+### Type `ScriptHashType`
+
+Specifies how the script `code_hash` is used to match the script code and how to run the code.
+
+Allowed kinds: "data", "type" and "data1".
+
+Refer to the section [Code Locating](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#code-locating) and [Upgradable Script](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0022-transaction-structure/0022-transaction-structure.md#upgradable-script) in the RFC *CKB Transaction Structure*.
+
+`ScriptHashType` is equivalent to `"data" | "type" | "data1"`.
+
+*   Type "data" matches script code via cell data hash, and run the script code in v0 CKB VM.
+*   Type "type" matches script code via cell type script hash.
+*   Type "data" matches script code via cell data hash, and run the script code in v1 CKB VM.
+
+
+### Type `JsonBytes`
+
+Variable-length binary encoded as a 0x-prefixed hex string in JSON.
+
+#### Example
+
+|  JSON | Binary |
+| --- |--- |
+|  “0x” | Empty binary |
+|  “0x00” | Single byte 0 |
+|  “0x636b62” | 3 bytes, UTF-8 encoding of ckb |
+|  “00” | Invalid, 0x is required |
+|  “0x0” | Invalid, each byte requires 2 digits |
+
+### Type `Versions`
+
+Version info of Godwoken & Web3 & Web3 Indexer.
+
+#### Examples
+
+
+```
+{
+  "web3Version": "1.0.0-rc1",
+  "web3IndexerVersion": "1.0.0-rc1",
+  "godwokenVersion": "1.1.0 f3cdd47"
+}
+```
+
+
+#### Fields
+
+`Versions` is a JSON object with the following fields.
+
+*   `web3Version`: `string` - Version of Godwoken Web3
+
+*   `web3IndexerVersion`: `string` - Version of Godwoken Web3 Indexer
+
+*   `godwokenVersion`: `string` - Version of Godwoken
+
+
+### Type `RollupCell`
+
+#### Fields
+
+`RollupCell` is a JSON object with the following fields.
+
+*   `typeHash`: [`H256`](#type-h256) - Hash of typeScript
+
+*   `typeScript`: [`Script`](#type-script) - Deploy info of rollup cell
+
+
+### Type `RollupConfig`
+
+#### Fields
+
+`RollupConfig` is a JSON object with the following fields.
+
+*   `requiredStakingCapacity`: [`Uint64`](#type-uint64) - Stacking capacity
+
+*   `challengeMaturityBlocks`: [`Uint64`](#type-uint64) - Challenge maturity blocks
+
+*   `finalityBlocks`: [`Uint64`](#type-uint64) - Finality Blocks
+
+*   `chainId`: [`Uint64`](#type-uint64) - Chain ID
+
+
+
+### Type `NodeInfo`
+
+Info of Godwoken & Web3 node.
+
+#### Fields
+
+`NodeInfo` is a JSON object with the following fields.
+
+*   `nodeMode`: `fullnode | readonly` - fullnode or readonly node
+
+*   `rollupCell`: [`RollupCell`](#type-rollupcell) - Rollup cell info
+
+*   `rollupConfig`: [`RollupConfig`](#type-rollupconfig) - Rollup config info
+
+*   `gwScripts`: [`GwScripts`](#type-gwscripts) - Godwoken scripts deploy info
+
+*   `eoaScripts`: [`EoaScripts`](#type-eoaScripts) - Supported EOA scripts
+
+*   `backends`: [`Backends`](#type-backends)
+
+*   `accounts`: [`Accounts`](#type-accounts) 
+
+*   `web3ChainId`: [`Uint64`](#type-uint64) - Chain ID of web3
```

Comment:
It should be the same with [chain id](https://github.com/nervosnetwork/godwoken-web3/pull/325/files#diff-1ccb96b1c0491eabd63696795735d87a29e86ad722db45fef58e90ed968261a7R528).

Later we can remove the prefix "web3" out, "chainId" is more clear.

Reply:
Updated